cinema/views.py

```diff
@@ -1 +1,70 @@
-# write views here
+from typing import Type
+
+from rest_framework import viewsets
+from rest_framework.serializers import Serializer
+
+from cinema.models import (
+    Movie,
+    Genre,
+    Actor,
+    MovieSession,
+    CinemaHall
+)
+from cinema.serializers import (
+    MovieSerializer,
+    GenreSerializer,
+    ActorSerializer,
+    MovieSessionSerializer,
+    CinemaHallSerializer,
+    MovieListSerializer,
+    MovieSessionListSerializer,
+    MovieSessionRetrieveSerializer,
+    MovieRetrieveSerializer
+)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_serializer_class(self) -> Type[Serializer]:
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+        return MovieSerializer
+
+    def get_queryset(self) -> queryset:
```

Comment:
The return type annotation for `get_queryset` should be `QuerySet` instead of `queryset`. Make sure to import `QuerySet` from Django's ORM if it's not already imported.